### PR TITLE
Issue #661 Add encrytion for plain text password

### LINKED ISSUE #661: Hostfolder configuration should not save the password in clear text
minishift hostfolder saves the password in clear text. So anyone who will get the access of  the workstation can view the file and can get read the password
```
$ cat ~/.minishift/config/allinstances.json 
{
	"HostFolders": [
		{
			"Name": "share",
			"Type": "cifs",
			"Options": {
				"domain": "test",
				"mountpoint": "",
				"password": "sample-password",
				"uncpath": "/Users/lmohanty/",
				"username": "lmohanty"
			}
		}
	]
```

### PR DESCRIPTION
**Pull request by praveenkumar**
Fix #661 

Please test it before merge. I also test it in my windows setup and if everything works then will remove `WIP` tag.

### CODE REVIEW

File: pkg/util/crypto.go [hunk 1]
```diff
@@ -89,3 +99,44 @@ func GenerateSelfSignedCert(certPath, keyPath string, ips []net.IP, alternateDNS
 
 	return nil
 }
+
+// Encrypt string to base64 crypto using AES
+func EncryptText(text string) (string, error) {
```
Comment (budhrg) [inline pkg/util/crypto.go]: @praveenkumar , I think we add unit test for this function

File: pkg/util/crypto.go [hunk 2]
```diff
@@ -89,3 +99,44 @@ func GenerateSelfSignedCert(certPath, keyPath string, ips []net.IP, alternateDNS
...
+
+	stream := cipher.NewCFBEncrypter(block, iv)
+	stream.XORKeyStream(ciphertext[aes.BlockSize:], plaintext)
+
+	// convert to base64
+	return base64.URLEncoding.EncodeToString(ciphertext), nil
+}
+
+// Decrypt from base64 to decrypted string
+func DecryptText(cryptoText string) (string, error) {
```
Comment (budhrg) [inline pkg/util/crypto.go]: Unit test is needed for this too

**Comment by budhrg** [comment]
@praveenkumar , my test result for Windows 10

```
PS C:\WINDOWS\system32> minishift hostfolder add budhrg
UNC path: \\192.168.1.105\C:\Users
Mountpoint [/mnt/sda1/budhrg]:
Username: budhram
Password: [HIDDEN]
Domain:
panic: crypto/aes: invalid key size 22

goroutine 1 [running]:
panic(0x9f9940, 0xc0422c2150)
        /usr/local/go/src/runtime/panic.go:500 +0x1af
github.com/minishift/minishift/pkg/util.EncryptText(0xc0422c20e8, 0x8, 0x0, 0x0, 0x0, 0x0)
        /home/minishift_ci/gopath/src/github.com/minishift/minishift/pkg/util/crypto.go:108 +0x2e6
github.com/minishift/minishift/pkg/minishift/hostfolder.Add(0xc04200a240, 0x6, 0x1, 0xc042112db0, 0x22)
        /home/minishift_ci/gopath/src/github.com/minishift/minishift/pkg/minishift/hostfolder/hostfolder.go:144 +0x2c3
github.com/minishift/minishift/cmd/minishift/cmd/hostfolder.glob..func1(0xe09da0, 0xc042114a50, 0x1, 0x1)
        /home/minishift_ci/gopath/src/github.com/minishift/minishift/cmd/minishift/cmd/hostfolder/add.go:47 +0x77
github.com/minishift/minishift/vendor/github.com/spf13/cobra.(*Command).execute(0xe09da0, 0xc042114a00, 0x1, 0x1, 0xe09da0, 0xc042114a00)
        /home/minishift_ci/gopath/src/github.com/minishift/minishift/vendor/github.com/spf13/cobra/command.go:650 +0x44a
github.com/minishift/minishift/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xe07ba0, 0x0, 0x0, 0x0)
        /home/minishift_ci/gopath/src/github.com/minishift/minishift/vendor/github.com/spf13/cobra/command.go:737 +0x36e
github.com/minishift/minishift/vendor/github.com/spf13/cobra.(*Command).Execute(0xe07ba0, 0xc0421cdf50, 0xc0421cdf50)
        /home/minishift_ci/gopath/src/github.com/minishift/minishift/vendor/github.com/spf13/cobra/command.go:695 +0x32
github.com/minishift/minishift/cmd/minishift/cmd.Execute()
        /home/minishift_ci/gopath/src/github.com/minishift/minishift/cmd/minishift/cmd/root.go:111 +0x34
main.main()
        /home/minishift_ci/gopath/src/github.com/minishift/minishift/cmd/minishift/main.go:26 +0x20
```

**Comment by LalatenduMohanty** [comment]
Getting the same error
```
C:\Users\lmohanty\Downloads\minishift-pr-762> minishift.exe hostfolder add home
UNC path: C:/Users/lmohanty
Mountpoint [/mnt/sda1/home]:
Username: lmohanty
Password: [HIDDEN]
Domain: WIN10
panic: crypto/aes: invalid key size 22

goroutine 1 [running]:
github.com/minishift/minishift/pkg/util.EncryptText(0xc0422668d0, 0xb, 0xc042266930, 0x5, 0x0, 0x0)
        C:/gopath/src/github.com/minishift/minishift/pkg/util/crypto.go:108 +0x2cb
github.com/minishift/minishift/pkg/minishift/hostfolder.Add(0xc04200c210, 0x4, 0x1, 0xc0420e0cf0, 0x23)
        C:/gopath/src/github.com/minishift/minishift/pkg/minishift/hostfolder/hostfolder.go:144 +0x18f
github.com/minishift/minishift/cmd/minishift/cmd/hostfolder.glob..func1(0xe66ba0, 0xc0422960a0, 0x1, 0x1)
        C:/gopath/src/github.com/minishift/minishift/cmd/minishift/cmd/hostfolder/add.go:47 +0x77
github.com/minishift/minishift/vendor/github.com/spf13/cobra.(*Command).execute(0xe66ba0, 0xc042296050, 0x1, 0x1, 0xe66ba0, 0xc042296050)
        C:/gopath/src/github.com/minishift/minishift/vendor/github.com/spf13/cobra/command.go:650 +0x238
github.com/minishift/minishift/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xe649a0, 0x0, 0xc04200dcc0, 0xc04200dcb0)
        C:/gopath/src/github.com/minishift/minishift/vendor/github.com/spf13/cobra/command.go:737 +0x340
github.com/minishift/minishift/vendor/github.com/spf13/cobra.(*Command).Execute(0xe649a0, 0xc0421c5f68, 0x9a84e9)
        C:/gopath/src/github.com/minishift/minishift/vendor/github.com/spf13/cobra/command.go:695 +0x32
github.com/minishift/minishift/cmd/minishift/cmd.Execute()
        C:/gopath/src/github.com/minishift/minishift/cmd/minishift/cmd/root.go:111 +0x34
main.main()
        C:/gopath/src/github.com/minishift/minishift/cmd/minishift/main.go:26 +0x2c
```

File: pkg/util/crypto.go (hunk 1 above)
Comment (hferentschik) [inline pkg/util/crypto.go]: +1

File: pkg/util/crypto.go (hunk 2 above)
Comment (hferentschik) [inline pkg/util/crypto.go]: +1

File: pkg/util/crypto.go (hunk 1 above)
Comment (hferentschik) [inline pkg/util/crypto.go]: this would probably have caught the Windows issue

**Comment by budhrg** [comment]
With update changes, it is working for me 👍 

```
PS C:\> cat C:\Users\budhram\.minishift\config\allinstances.json
{
        "HostFolders": [
                {
                        "Name": "budhrg",
                        "Type": "cifs",
                        "Options": {
                                "domain": "",
                                "mountpoint": "",
                                "password": "Kxiz-P9IiChM8D3ZTFdnTEnDhj1o44KH",
                                "uncpath": "//10.70.35.155/Users",
                                "username": "budhram"
                        }
                }
        ]
}
```
Also able to mount `C:\Users` folder . (Note: C:\Users should be shared with `Share -> Advanced Sharing`

**Comment by hferentschik** [comment]
Rebased and merged, thanks.

**Pull request closed without merging**